### DISCUSSION TITLE
How to get params of optimizer?

### DISCUSSION THREAD

**Discussion by arihantbansal** [google-deepmind/optax#316] [Q&A] [answered]
Is there any way to get parameters of the optimizer/network at a step given the optimizer state? Similar to `get_params(opt_state)` which exists in [`jax.example_libraries.optimizers`](https://github.com/google/jax/blob/main/jax/example_libraries/optimizers.py)

**Comment by pharringtonp19** [comment, marked as answer]
@arihantbansal  

I think you are looking for the following two lines of code.

```
updates, opt_state = optimizer.update(grads, opt_state)
params = optax.apply_updates(params, updates)
```
At each iteration

1. Update the optimizer state `optimizer.update(grads, opt_state)`
2. Update the parameters of the neural network `optax.apply_updates(params, updates)`

**Reply by louiskenyon98** [reply]
How do you get the initial optimizer state, given parameters, for the network? When initialising with the previous method in Jax you would call get_params() to obtain the starting state of the network which is then passed to the step function.

I found my own answer: you call ```optimizer.init(params)```
